hydroflow/src/builder/hydroflow_builder.rs

```diff
@@ -132,6 +133,13 @@ impl HydroflowBuilder {
         self.hydroflow
     }
 
+    pub fn start_iter<I>(&self, iter: I) -> IterPullSurface<I::IntoIter>
```

Comment:
perhaps we can make this a trait/method on iterator so you could do like, `iter.hydrate()` (but probably with a better name than that)